### PR TITLE
Update ASM to 9.7 for plugin scanner

### LINKED ISSUE #108776: [CI] Multiple test failures JDK23 on main
Multiple test failures due to "Unsupported class file major version 67".

e.g. https://gradle-enterprise.elastic.co/s/3vr5hou5vgyzc/tests/task/:libs:elasticsearch-plugin-scanner:test/details/org.elasticsearch.plugin.scanner.NamedComponentScannerTests/testNamedComponentsAreFoundWhenSingleJarProvided?top-execution=1

```
./gradlew ':libs:elasticsearch-plugin-scanner:test' --tests "org.elasticsearch.plugin.scanner.NamedComponentScannerTests.testNamedComponentsAreFoundWhenSingleJarProvided" -Dtests.seed=8BC38CBC63E90492 -Dtests.locale=es -Dtests.timezone=America/Indiana/Winamac -Druntime.java=23
```

But there are plenty of others popping up atm, e.g.

[org.elasticsearch.plugin.scanner.NamedComponentScannerTests](https://gradle-enterprise.elastic.co/s/3vr5hou5vgyzc/tests/task/:libs:elasticsearch-plugin-scanner:test/details/org.elasticsearch.plugin.scanner.NamedComponentScannerTests)
[testNamedComponentsAreFoundWhenSingleJarProvided](https://gradle-enterprise.elastic.co/s/3vr5hou5vgyzc/tests/task/:libs:elasticsearch-plugin-scanner:test/details/org.elasticsearch.plugin.scanner.NamedComponentScannerTests/testNamedComponentsAreFoundWhenSingleJarProvided)
[testNamedComponentsCanExtednCommonSuperClass](https://gradle-enterprise.elastic.co/s/3vr5hou5vgyzc/tests/task/:libs:elasticsearch-plugin-scanner:test/details/org.elasticsearch.plugin.scanner.NamedComponentScannerTests/testNamedComponentsCanExtednCommonSuperClass)
[org.elasticsearch.plugin.scanner.ClassReadersTests](https://gradle-enterprise.elastic.co/s/3vr5hou5vgyzc/tests/task/:libs:elasticsearch-plugin-scanner:test/details/org.elasticsearch.plugin.scanner.ClassReadersTests)
[testMultipleJarsInADir](https://gradle-enterprise.elastic.co/s/3vr5hou5vgyzc/tests/task/:libs:elasticsearch-plugin-scanner:test/details/org.elasticsearch.plugin.scanner.ClassReadersTests/testMultipleJarsInADir)
[testStreamOfJarsAndIndividualClasses](https://gradle-enterprise.elastic.co/s/3vr5hou5vgyzc/tests/task/:libs:elasticsearch-plugin-scanner:test/details/org.elasticsearch.plugin.scanner.ClassReadersTests/testStreamOfJarsAndIndividualClasses)
[testTwoClassesInAStreamFromJar](https://gradle-enterprise.elastic.co/s/3vr5hou5vgyzc/tests/task/:libs:elasticsearch-plugin-scanner:test/details/org.elasticsearch.plugin.scanner.ClassReadersTests/testTwoClassesInAStreamFromJar)

Reproduces locally for me with Open JDK "23.ea.22-open"

### PR DESCRIPTION
**Pull request by rjernst**
This commit updates the ASM library in order to support class files written with Java 23.

closes #108776

### CODE REVIEW

File: gradle/verification-metadata.xml [hunk 1]
```diff
@@ -4107,6 +4112,11 @@
             <sha256 value="c43ecf17b539c777e15da7b5b86553b377e2d39a683de6285567d5283888e7ef" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="org.ow2.asm" name="asm-tree" version="9.7">
```
Comment (pgomulka) [inline gradle/verification-metadata.xml:4115]: we should probably remove the previous entries up to this version

**Review by pgomulka — APPROVED** [review]
LGTM,
should we remove previous entries in verification-metadata?

File: gradle/verification-metadata.xml (hunk 1 above)
Comment (rjernst) [reply gradle/verification-metadata.xml:4115]: Older asm versions are still used by other modules (eg painless is on an older asm). We should look at cleaning up verification metadata separately, I think it is a gradle/build concern that needs to be handled consistently.

Comment (rjernst) [reply gradle/verification-metadata.xml:4115]: I opened https://github.com/elastic/elasticsearch/issues/108825

**Pull request merged**